src/applications/vaos/services/patient/index.js

```diff
@@ -0,0 +1,388 @@
+    recordVaosError(`eligibility-${errorKey}`);
+    return new Error('Eligibility error');
+  };
+}
+
+const PRIMARY_CARE = '323';
+
+const DISABLED_LIMIT_VALUE = 0;
+
+function hasVisitedInPastMonths(pastVisit) {
+  if (!pastVisit) {
+    return true;
+  }
+
+  return (
+    pastVisit.durationInMonths === DISABLED_LIMIT_VALUE ||
+    pastVisit.hasVisitedInPastMonths
+  );
+}
+
+function isUnderRequestLimit(data) {
+  return (
+    data?.requestLimit === DISABLED_LIMIT_VALUE ||
+    data?.numberOfRequests < data?.requestLimit
+  );
+}
+
+async function fetchPatientEligibilityFromVAR({
+  typeOfCare,
+  location,
+  type = null,
+}) {
+  const checks = {};
+  if (type !== 'direct') {
+    checks.requestLimit = getRequestLimits(location.id, typeOfCare.id)
+      .then(resp => resp[0])
+      .catch(createErrorHandler('request-exceeded-outstanding-requests-error'));
+  }
+
+  if (type !== 'direct' && typeOfCare.id !== PRIMARY_CARE) {
+    checks.requestPastVisit = checkPastVisits(
+      location.vistaId,
+      location.id,
+      typeOfCare.id,
+      'request',
+    ).catch(createErrorHandler('request-check-past-visits-error'));
+  }
+
+  if (type !== 'request' && typeOfCare.id !== PRIMARY_CARE) {
+    checks.directPastVisit = checkPastVisits(
+      location.vistaId,
+      location.id,
+      typeOfCare.id,
+      'direct',
+    ).catch(createErrorHandler('direct-check-past-visits-error'));
+  }
+
+  const results = await promiseAllFromObject(checks);
+  const output = { direct: null, request: null };
+
+  if (results.directPastVisit instanceof Error) {
+    output.direct = new Error('Direct scheduling eligibility check error');
+  } else if (type !== 'request') {
+    output.direct = {
+      hasRequiredAppointmentHistory: hasVisitedInPastMonths(
+        results.directPastVisit,
+      ),
+    };
+  }
+
+  if (
+    results.requestPastVisit instanceof Error ||
+    results.requestLimit instanceof Error
+  ) {
+    output.request = new Error('Request eligibility check error');
+  } else if (type !== 'direct') {
+    output.request = {
+      hasRequiredAppointmentHistory: hasVisitedInPastMonths(
+        results.requestPastVisit,
+      ),
+      isEligibleForNewAppointmentRequest: isUnderRequestLimit(
+        results.requestLimit,
+      ),
+    };
+  }
+
+  return output;
+}
+
+/**
+ * Returns patient based eligibility checks for specified request or direct types
+ *
+ * @export
+ * @param {Object} params
+ * @param {TypeOfCare} params.typeOfCare Type of care object,
+ * @param {Location} params.location Location of where patient should have eligibility checked,
+ * @param {'direct'|'request'|null} [params.type=null] The type to check eligibility for. By default,
+ *   will check both
+ * }
+ * @returns {PatientEligibility} Patient eligibility data
+ */
+export async function fetchPatientEligibility({
+  typeOfCare,
+  location,
+  type = null,
+}) {
+  return fetchPatientEligibilityFromVAR({ typeOfCare, location, type });
```

Comment:
We will have a feature flag switch in here when we integrated the patient metadata endpoint from the vaos service.